File: Java/Fundamentals/Generic/Collections/Interval.java
class Interval implements Comparable<Interval>{

	private int min, sec;

	public Interval(int m, int s){
		min = m + s / 60;
		sec = s % 60;
	}

	public int minutes(){
		return min;
	}

	public int seconds(){
		return sec;
	}

	public int time(){
		return 60 * min + sec;
	}

	public String toString(){
		return String.format("%d:%02d", min, sec);
	}

	public int hashCode(){
		return 1000 * time();
	}

	public boolean equals(Object other){
		if(other instanceof Interval){
			Interval that = (Interval)other;
			return (this.min == that.min) && (this.sec == that.sec);
		}
		return false;
	}

	public int compareTo(Interval that){
		return this.time() - that.time();
	}

	public static final Interval ZERO = new Interval(0, 0);

	public Interval add(Interval that){
		return new Interval(this.min + that.min, this.sec + that.sec);
	}
}

File: MS.NET/Fundamentals/Generics/Basics/interval.cs
using System;

partial class Interval : IComparable<Interval>
{
	private int min, sec;

	partial void OnInitialize();

	public Interval(int m, int s)
	{
		min = m + s / 60;
		sec = s % 60;
		OnInitialize();
	}

	public int Minutes => min;

	public int Seconds => sec;

	public int Time => 60 * min + sec;
	
	public override string ToString()
	{
		return $"{min}:{sec:00}"; //string.Format("{0}:{1:00}", min, sec)
	}

	public override int GetHashCode()
	{
		return 1000 * Time;
	}

	public override bool Equals(object other)
	{
		if(other is Interval that)
			return (this.min == that.min) && (this.sec == that.sec);
		
		return false;
	}

	public int CompareTo(Interval that)
	{
		return this.Time - that.Time;
	}
		
}
